### PR TITLE
Refactor provider storage

### PR DESCRIPTION
**Pull request by oliviermichaelis**
* [breaking] The storage package expects another object store layout for providers
  The full prefix will be `<bucketPrefix>/providers/internal/<namespace>/<name>/`
  
I tested the S3 implementation on MinIO, but didn't test it for GCS

### CODE REVIEW

File: pkg/storage/collection.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package storage
```
Comment (oliviermichaelis) [inline pkg/storage/collection.go:1]: This is a 1to1 copy of what was previously in the provider package. Nothing new

File: pkg/storage/errors.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package storage
+
+import "errors"
+
+// Storage errors.
+var (
+	ErrAlreadyExists = errors.New("provider already exists")
+	ErrNotFound      = errors.New("failed to locate provider")
```
Comment (oliviermichaelis) [inline pkg/storage/errors.go:8]: The first two errors are not used yet

File: pkg/storage/gcs.go [hunk 3]
```diff
@@ -11,106 +11,68 @@ import (
...
 )
 
+// GCSStorage implements provider.Storage
 type GCSStorage struct {
 	sc              *storage.Client
 	bucket          string
 	bucketPrefix    string
 	useSignedURL    bool
-	signedURLExpiry int64
+	signedURLExpiry time.Duration
```
Comment (oliviermichaelis) [inline pkg/storage/gcs.go:27]: I refactored this to use time.Duration instead of int64

**Review by heimweh — APPROVED** [review]
Nice work! 🎉 This LGTM 🙏

**Pull request merged**